File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/OpenIPC2.pl
#!/usr/bin/perl -w

### OpenIPC2.pl

use strict;                                                        #(P)

print "Started OpenIPC2 process with PID $$\n";                    #(Q)

while (<>) {                                                       #(R)
    print "MESSAGE RECEIVED FROM PIPELINE: $_";                    #(S)
}

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/OpenIPC2.py
#!/usr/bin/python

### OpenIPC2.py

import os                            # for getpid()                 #(L)
import sys                           # for stdin                    #(M)

print "Started OpenIPC2 child process with PID", os.getpid()        #(N)

while ( True ):                                                     #(O)
    input = sys.stdin.readline()                                    #(P)
    if (input == ''): break                                         #(Q)
    print  "RECEIVED BY CHILD FROM PIPELINE: ", input               #(R)
